src/lib/gossip_net/gossip_net.ml

```diff
@@ -356,7 +356,23 @@ module Make (Message : Message_intf) : S with type msg := Message.msg = struct
                   (Hashtbl.find t.connections client_inet_addr)
                   ~f:Fn.id
               in
-              if
+              let is_client_banned =
+                let peer_status =
+                  Trust_system.Peer_trust.lookup t.trust_system
+                    client_inet_addr
+                in
+                match peer_status.banned with
+                | Banned_until _ ->
+                    true
+                | Unbanned ->
+                    false
+              in
+              if is_client_banned then (
+                Logger.error t.logger ~module_:__MODULE__ ~location:__LOC__
+                  "Disconnecting banned peer %s"
```

Comment:
This isn't actually an error. Should be info level I think. And can the message be "rejecting connection from banned peer" or something along those lines to make it clear the difference between this and closing connections that were open when the peer was banned?